api/demo/demo.go

```diff
@@ -0,0 +1,87 @@
+package demo
+
+import (
+	"log"
+
+	"github.com/pkg/errors"
+	portainer "github.com/portainer/portainer/api"
+	"github.com/portainer/portainer/api/dataservices"
+)
+
+type EnvironmentDetails struct {
+	Enabled      bool                   `json:"enabled"`
+	Users        []portainer.UserID     `json:"users"`
+	Environments []portainer.EndpointID `json:"environments"`
+}
+
+type Service struct {
+	details EnvironmentDetails
+}
+
+func NewService() *Service {
+	return &Service{}
+}
+
+func (service *Service) Details() EnvironmentDetails {
+	return service.details
+}
+
+func (service *Service) Init(store dataservices.DataStore, cryptoService portainer.CryptoService) error {
+	log.Print("[INFO] [main] Starting demo environment")
+
+	err := initDemoUser(store, cryptoService)
+	if err != nil {
+		return errors.WithMessage(err, "failed creating demo user")
+	}
+
+	err = initDemoLocalEndpoint(store)
+	if err != nil {
+		return errors.WithMessage(err, "failed creating demo endpoint")
+	}
+
+	err = initDemoSettings(store)
+	if err != nil {
+		return errors.WithMessage(err, "failed updating demo settings")
+	}
+
+	service.details = EnvironmentDetails{
+		Enabled: true,
+		Users:   []portainer.UserID{1},
+		// endpoints 2,3 are created after deployment of portainer
+		Environments: []portainer.EndpointID{1, 2, 3},
```

Comment:
Still not sure where 2 and 3 come from, could you please elaborate on how they'd be added?
If we expect another 2 endpoints be created after the first one and there is no way to create their objects as a part of this `Init` fn, we could still get the first id from the sequence (N) and assume N+1 and N+2.

Reply:
they're added here https://github.com/portainer/portainer-demo/blob/master/docker-compose.yml

I would also prefer to either add them here (in code), or to make it possible to add "demo" envs from the API. 

I didn't find it easy to add them in codebase.

I decided to make it this way because I wanted to merge this PR fast so demo branch will stop diverging from develop.

Here's a ticket I opened for it https://portainer.atlassian.net/browse/EE-2186

Reply:
Interesting, thanks for pointing it out to me 👍 
Also support the idea of "readonly" or "admin-edit-only" endpoints, maybe they find reasons to exist outside of demo scope.
Do we expect users to run Portainer in demo mode? If so, they won't use our demo docker-compose but they still won't be able to remove endpoints.

Reply:
> Also support the idea of "readonly" or "admin-edit-only" endpoints, maybe they find reasons to exist outside of demo scope.

while developing this feature I felt refactoring to add "readonly" endpoints, added too much noise to this PR and it should be done in another ticket.

> Do we expect users to run Portainer in demo mode? If so, they won't use our demo docker-compose but they still won't be able to remove endpoints.

I don't think so. this is only for demo.portainer.io

Reply:
> while developing this feature I felt refactoring to add "readonly" endpoints, added too much noise to this PR and it should be done in another ticket.
yep, makes sense 👍